core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -205,18 +205,18 @@ public void commit() throws IOException {
             try {
                 Files.move(tmpPath, destPath, StandardCopyOption.REPLACE_EXISTING);
             } catch (IOException e1) {
-                e1.addSuppressed(e);
+                e.addSuppressed(e1);
```

Comment:
re-assigning a exception defined in a catch makes the code harder to follow for me.

Reply:
OK, I'm confident the pull request can be reworked to not reassign the exception.  I am not ready to attempt that type of rework from the GitHub suggestions, so it will need to wait for @olivergondza to decide if he'd like to proceed with this pull request, adapt it based on code review, or close it.